Improve docs for remote storage (#53)
PrefectHQ/prefect-ray: README.md

When using the `RayTaskRunner` with a remote Ray cluster, you may run into issues that are not seen when using a local Ray instance. To resolve these issues, we recommend taking the following steps when working with a remote Ray cluster:

1. By default, Prefect will not persist any data to the filesystem of the remote ray worker. However, if you want to take advantage of Prefect's caching ability, you will need to configure a remote result storage and persist the results, like shown below:

> ```bash

Author: ```suggestion
```python
```

from typing import List

from prefect import flow, get_run_logger, task
from prefect.filesystems import S3
from prefect.tasks import task_input_hash
from prefect_ray.task_runners import RayTaskRunner

@task(cache_key_fn=task_input_hash)
def say_hello(name: str) -> None:
    logger = get_run_logger()

>     # this should only show if not cached

Reviewer: ```suggestion
    # This log statement will print only on the first run. Subsequent runs will be cached.
```

    logger.info(f"hello {name}!")
    return name

@flow(
    task_runner=RayTaskRunner(
        address="ray://<instance_public_ip_address>:10001",
    ),

>     persist_result=True,

Reviewer: I don't think this line is strictly necessary for this example, but I also don't think there's any harm in leaving it in.

Reviewer 2: I think it could be confusing to include it, it's not necessary so I would not.

>     result_storage=S3(bucket_path="<bucket_path>"),

Reviewer: I think we should encourage people to create their storage via the UI.
```suggestion
    # Using an S3 block that has already been created via the Prefect UI
    result_storage=S3.load("my_results_storage"),
```

Reviewer 2: I'd use a slug there so it's loaded at runtime instead of script execution, e.g. `s3/my_result_storage`

)
def greetings(names: List[str]) -> None:
    for name in names:
        say_hello.submit(name)

if __name__ == "__main__":
    greetings(["arthur", "trillian", "ford", "marvin"])
```

2. If you get an error stating that the module 'prefect' cannot be found, ensure `prefect` is installed on the remote cluster, with:
```bash
pip install prefect
```

> 3. If you get an error stating something like "File system created with scheme 's3' could not be created", double check that the required Python modules are installed on **both local and remote machines**. For example, if using S3 for the remote storage:

Reviewer: ```suggestion
3. If you get an error with a message similar to "File system created with scheme 's3' could not be created", ensure the required Python modules are installed on **both local and remote machines**. The required prerequisite modules can be found in the [Prefect documentation](https://docs.prefect.io/tutorials/storage/#prerequisites). For example, if using S3 for the remote storage:
```

1. By default, Prefect will not persist any data to the filesystem of the remote ray worker. However, if you want to take advantage of Prefect's caching ability, you will need to configure a remote result storage to persist results across task runs. 

We recommend using the [Prefect UI to configure a storage block](https://docs.prefect.io/ui/blocks/) to use for remote results storage.

Here's an example of a flow that uses caching and remote result storage:
```python
from typing import List

from prefect import flow, get_run_logger, task
from prefect.filesystems import S3
from prefect.tasks import task_input_hash
from prefect_ray.task_runners import RayTaskRunner

# The result of this task will be cached in the configured result storage
@task(cache_key_fn=task_input_hash)
def say_hello(name: str) -> None:
    logger = get_run_logger()
    # This log statement will print only on the first run. Subsequent runs will be cached.
    logger.info(f"hello {name}!")
    return name

@flow(
    task_runner=RayTaskRunner(
        address="ray://<instance_public_ip_address>:10001",
    ),
    # Using an S3 block that has already been created via the Prefect UI

>     result_storage="s3/my_result_storage",

Reviewer: Needs to have dashes to be a valid block document name
```suggestion
    result_storage="s3/my-result-storage",
```

Author: Thanks!
